source/slang/slang-ir-insts.h

```diff
@@ -327,6 +327,18 @@ struct IRRequireGLSLVersionDecoration : IRDecoration
     }
 };
 
+struct IRSPIRVNonUniformResourceDecoration : IRDecoration
+{
+    enum { kOp = kIROp_SPIRVNonUniformResourceDecoration };
+    IR_LEAF_ISA(RequireGLSLVersionDecoration)
+
+        IRConstant* getSPIRVVersionOperand() { return cast<IRConstant>(getOperand(0)); }
+    IntegerLiteralValue getSPIRVVersion()
+    {
+        return getSPIRVVersionOperand()->value.intVal;
+    }
```

Comment:
How is getSPIRVVersion used?
If it is just to get a SPIR-V version, this can be acquired from `linkage->targets[i]->getTargetCaps()`, where getTargetCaps() returns a capability set with the SPIR-V version being targeted (or at-least is supposed to at some point)

Reply:
I updated this struct to return the nonuniform resource value as it is typically an integer index.